### PR TITLE
[5409] Publishing Dashboard in Project Tools not showing Unlock Dialog. Also Publishing Status dialog not updating correctly

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5409

### CODE REVIEW

File: ui/app/src/components/PublishOnDemandWidget/PublishOnDemandWidget.tsx [hunk 1]
```diff
@@ -201,7 +201,7 @@ export function PublishOnDemandWidget(props: PublishOnDemandWidgetProps) {
             message: formatMessage(messages.publishSuccess)
           })
         );
-        setPublishGitFormData(initialPublishGitFormData);
+        setPublishGitFormData({ ...initialPublishGitFormData, environment });
```
Comment (rart) [inline ui/app/src/components/PublishOnDemandWidget/PublishOnDemandWidget.tsx]: Can we rename to `publishingTarget` with ease?

File: ui/app/src/components/DialogHeaderAction/DialogHeaderAction.tsx [hunk 2]
```diff
@@ -18,15 +18,20 @@ import IconButton, { IconButtonProps } from '@mui/material/IconButton';
...
+  confirmProps?: ConfirmDropdownProps;
 }
 
 export function DialogHeaderAction(props: DialogHeaderActionProps) {
-  const { icon, tooltip, disabled = false, ...rest } = props;
-  return tooltip ? (
+  const { icon, tooltip, disabled = false, confirmation = false, confirmProps, ...rest } = props;
+  return confirmation ? (
+    <ConfirmDropdown {...confirmProps} />
+  ) : tooltip ? (
```
Comment (rart) [inline ui/app/src/components/DialogHeaderAction/DialogHeaderAction.tsx]: This makes the DialogHeaderAction work in two very different directions. Rollback and use the click of the action show what's needed.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/PublishOnDemandWidget/PublishOnDemandWidget.tsx (hunk 1 above)
Comment (jvega190) [reply ui/app/src/components/PublishOnDemandWidget/PublishOnDemandWidget.tsx]: done!

File: ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx [hunk 3]
```diff
@@ -96,7 +96,7 @@ export function PublishCommitDialog(props: PublishCommitDialogProps) {
         const newData: Partial<PublishCommitDialogState> = { publishingTargets, loadingPublishingTargets: false };
         // Set pre-selected environment.
```
Comment (rart) [inline ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx]:
```suggestion
        // Set pre-selected publishing target.
```

**Review by rart — APPROVED** [review]

**Pull request merged**